src/Microsoft.ML.DataView/VBuffer.cs

```diff
@@ -251,16 +310,34 @@ public static void Copy(T[] src, int srcIndex, ref VBuffer<T> dst, int length)
             dst = editor.Commit();
         }
 
+        /// <summary>
+        /// Returns the joint list of all index/value pairs.
+        /// </summary>
+        /// <param name="all">If <see langword="true"/> all pairs, even those implicit values of a sparse representation,
+        /// will be returned, with the implicit values having the default value, as is appropriate. If left
+        /// <see langword="false"/> then only explicitly defined values are returned.</param>
+        /// <returns></returns>
         public IEnumerable<KeyValuePair<int, T>> Items(bool all = false)
         {
             return Items(_values, _indices, Length, _count, all);
         }
 
+        /// <summary>
+        /// Returns an enumerable with <see cref="Length"/> items, representing the values.
+        /// </summary>
         public IEnumerable<T> DenseValues()
         {
             return DenseValues(_values, _indices, Length, _count);
         }
 
+        /// <summary>
+        /// Gets the item stored in this structure. In the case of a dense vector this is a simple lookup.
+        /// In the case of a sparse vector, it will try to find the entry with that index, and set <paramref name="dst"/>
+        /// to that stored value, or if no such value was found, assign it the default value.
+        /// </summary>
+        /// <param name="slot">The slot index, which must be a non-negative number less than <see cref="Length"/>.
+        /// If it is not in that </param>
+        /// <param name="dst">The value stored </param>
```

Comment:
>d  [](start = 45, length = 2)

oh no! missing Dot!

Reply:
Missing a sentence more like.

Reply:
I think I was going to write something about explicit and implicit values, went off and wrote that on the class documentation so that it was a defined term, but then forgot to come back here to make use of that clarifying note. 😄